lib/tictactoe/validation.ex

```diff
@@ -0,0 +1,21 @@
+defmodule TicTacToe.Validation do
+  alias TicTacToe.Board
+  alias TicTacToe.Game
+
+  def check_selection(selection, %Game{board: %Board{} = board, markers: markers} = game) do
+    cond do
+      invalid_character?(selection, game) -> {:char, game}
+      Board.space_occupied?(to_index(selection), board, markers) -> {:occupied, game}
```

Comment:
Let's try to follow [the convention](https://elixirschool.com/en/lessons/intermediate/error_handling#general-conventions-0) when it comes to errors